simphony/testing/abc_check_engine.py

```diff
@@ -212,6 +215,7 @@ def test_dataset_rename(self):
         # we should be able to access using the new "bar" name
         ds_bar = engine.get_dataset("bar")
         self.assertEqual(ds_bar.name, "bar")
+        engine.remove_dataset("bar")
```

Comment:
The point is that engines that only support a single dataset will not allow adding another dataset with the name "foo" (or with any other name either).